### ISSUE TITLE
Build Project. 2gb memory is not enough

### ISSUE THREAD

**Issue by b01110011** [nuxt/nuxt.js#8080]
### Versions

- nuxt: v2.14.0
- node: v12.16.0

I have VDS server with 2gb memory.
When I run build project, I get memory error.
Why 2gb memory is not enough? May be I do something wrong?
Launched project takes 80mb memory.

![Screenshot_7](https://user-images.githubusercontent.com/39577128/93204833-f5353180-f780-11ea-9bef-afa833a45743.png)
![Screenshot_8](https://user-images.githubusercontent.com/39577128/93204837-f5cdc800-f780-11ea-9686-545a1b687b81.png)
![Screenshot_9](https://user-images.githubusercontent.com/39577128/93204839-f6665e80-f780-11ea-84fd-1fe1fd6a44d8.png)
![Screenshot_10](https://user-images.githubusercontent.com/39577128/93204840-f6fef500-f780-11ea-9795-817d0424ae1e.png)
![Screenshot_11](https://user-images.githubusercontent.com/39577128/93205132-873d3a00-f781-11ea-83d3-229ee924096c.png)

**Comment by b01110011** [comment]
I try to run that `node --max-old-space-size=800 node_modules\nuxt\bin\nuxt.js build` but anyway I get memory error.

**Comment by jonasolesen** [comment]
When I build my application, the memory usage peaks at around 1 GB. Have you tried just cranking the memory to 1.8-2 GB?

**Comment by b01110011** [comment]
Yes, It works.
`node --max-old-space-size=1600 node_modules/nuxt/bin/nuxt.js build`

but I would like it to work out of the box.
Something like `nuxt build --max-old-space-size=1600` or `nuxt build --max-memory=1600`

**Comment by jonasolesen** [comment]
Why not just add it as a script in your package.json file?

```
"scripts": {
    "build": "node --max-old-space-size=1600 node_modules/nuxt/bin/nuxt.js build"
}
```

Then it's just a matter of running `npm run build` or `yarn build`.

**Comment by b01110011** [comment]
In windows os path will be `node_modules\\nuxt\\bin\\nuxt.js`

**Comment by gryphonmyers** [comment]
Related https://github.com/nuxt/nuxt.js/issues/7855

**Comment by rosnaib11** [comment]
Very similar issue to me, I got 1gb ram VPS server and didn't work on my universal app. So I upgrade it to 2gb ram, it works couple of times but soon as my website grows I started experiencing the error again. I wrote this [blog] that I used as alternative methods instead of upgrading my ram again. (https://www.utmostcontent.com/posts/21/nuxt-js-%22npm-run-build%22-javascript-heap-out-of-memory-error)
